drivers/tsl2561/tsl2561.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (C) 2016 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_tsl2561
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the TSL2561 Luminosity sensor.
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <math.h>
+#include <string.h>
+
+#include "log.h"
+#include "tsl2561.h"
+#include "tsl2561_internals.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/* internal helpers */
+static void _enable(tsl2561_t *dev);
+static void _disable(tsl2561_t *dev);
+static void _read_data(tsl2561_t *dev, uint16_t *full, uint16_t *ir);
+
+#if ENABLE_DEBUG
+static void _print_init_info(tsl2561_t *dev)
+{
+    DEBUG("[Info] I2C device: %d\n", dev->i2c_dev);
+    DEBUG("[Info] Address: %d\n", dev->addr);
+    switch(dev->gain) {
+    case TSL2561_GAIN_1X:
```

Comment:
Would you indent the `case` please? I'm not 100% sure we have this in the coding conventions but it is "common" for RIOT.